src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -216,6 +221,10 @@ private void validateParameters(Map<String, Object> parameterMap) {
         checkRequiredParameterPresent(parameterMap, errors, IOT_DATA_ENDPOINT_PARAMETER_NAME);
         checkRequiredParameterPresent(parameterMap, errors, PROVISIONING_TEMPLATE_PARAMETER_NAME);
         checkRequiredParameterPresent(parameterMap, errors, ROOT_PATH_PARAMETER_NAME);
+        if (!(parameterMap.get(CSR_PATH_PARAMETER_NAME) == null
+                || Utils.isEmpty(parameterMap.get(CSR_PATH_PARAMETER_NAME).toString()))) {
+            checkRequiredParameterPresent(parameterMap, errors, CSR_PRIVATE_KEY_PATH_PARAMETER_NAME);
+        }
```

Comment:
Don't we want to be smarter about requiring claim cert and key params with this change now?

Reply:
Ideally, yes. However, provisioning still requires the claim cert and key params while building a mqtt connection.
https://github.com/aws-greengrass/aws-greengrass-fleet-provisioning-by-claim/blob/main/src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java#L131-L152
I was hoping to find a way to build a connection using csr but that is not available

Reply:
That connection doesn't seem useful if you're trying to provision with a CSR though, so that code path should be bypassed. It's not going to be a nice user experience to require these params in config file when they're not technically needed

Reply:
Actually that connection is being used to fetch the csr response from iot core
https://github.com/aws-greengrass/aws-greengrass-fleet-provisioning-by-claim/blob/main/src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java#L173-L175

Reply:
The CSR and certificate are quite different. The certificate used is the claim cert. The CSR just allows you to specify how the cert+key will be generated by iot core when you provision.

Reply:
Should've looked closely